sequence_task_3/student/src/impl.cpp

```diff
@@ -1,2 +1,26 @@
 #include "impl.h"
+#include <algorithm>
+
+std::string reverseWords(const std::string& input)
+{
+    std::stringstream istream(input);
+    std::deque<std::string> deque;
+
+    std::for_each(std::istream_iterator<std::string>(istream), std::istream_iterator<std::string>(), [&deque]( const std::string& s )
+    {
+        deque.push_front(s);
+    });
+
+    std::stringstream outstream;
+    std::for_each(deque.begin(), deque.end(), [&outstream, &deque]( const std::string& word )
+    {
+        outstream << word;
+        if (word != deque.back())
```

Comment:
add a test case with duplicated words and see how string equality operator will work here :)